File: test-suite/src/test/java/io/micronaut/docs/http/client/bind/type/MetadataClient.java
package io.micronaut.docs.http.client.bind.type;

import io.micronaut.docs.http.client.bind.type.Metadata;
import io.micronaut.http.annotation.Get;
import io.micronaut.http.client.annotation.Client;

//tag::clazz[]
@Client("/")
public interface MetadataClient {

    @Get("/client/bind")
    String get(Metadata metadata);
}
//end::clazz[]

File: test-suite-kotlin/src/test/kotlin/io/micronaut/docs/http/client/bind/annotation/MetadataClient.kt
package io.micronaut.docs.http.client.bind.annotation

import io.micronaut.http.annotation.Get
import io.micronaut.http.client.annotation.Client

//tag::clazz[]
@Client("/")
interface MetadataClient {

    @Get("/client/bind")
    operator fun get(@Metadata metadata: Map<String, Any>): String
}
//end::clazz[]
